src/TRoomDB.cpp

```diff
@@ -65,13 +65,12 @@ bool TRoomDB::addRoom(int id)
         rooms[id]->setId(id);
         // there is no point in updating the entranceMap here, as the room has no exit information
         return true;
-    } else {
-        if (id <= 0) {
-            QString error = qsl("addRoom: illegal room id=%1. roomID must be > 0").arg(id);
-            mpMap->logError(error);
-        }
-        return false;
     }
+    if (id <= 0) {
+        QString error = qsl("addRoom: illegal room id=%1. roomID must be > 0").arg(id);
+        mpMap->logError(error);
+    }
+    return false;
```

Comment:
We return `false` if the room is not created - but we do not report here if the room already exists AND *that* is why this method failed. Perhaps we review the usage of this method and determine whether the reporting of failures need to be unified/be made more rational?

Reply:
Sure, could be a good improvement but it is not in the scope of speeding up room creation.